internal/compatibility/compatibility.go

```diff
@@ -149,6 +149,16 @@ func NextMinorVersion(version string) (string, error) {
 	return fmt.Sprintf("v%d.%d", major, minor+1), nil
 }
 
+// PriorMinorVersion returns the prior minor version for a given canonical semver.
+// The returned format is vMAJOR.MINOR.
+func PriorMinorVersion(version string) (string, error) {
```

Comment:
nit: inline the functionality of this function in `cluster_resource_test.go`, since that is the only place it is used (don't export other functions, just copy the needed functionality).